### PR TITLE
AEA-1239 / APM-1486: Switch to using egress proxy for Spine and ODS communication.

### PR DESCRIPTION
**Pull request by vurt007**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,13 +26,13 @@ release:
...
-		cat ecs-proxies-deploy.yml | sed -e 's/{{ SPINE_ENV }}/test/g' | sed -e 's/{{ SANDBOX_MODE_ENABLED }}/1/g' > dist/ecs-deploy-$$env.yml; \
+		cat ecs-proxies-deploy.yml | sed -e 's/{{ SPINE_ENV }}/veit07/g' | sed -e 's/{{ SANDBOX_MODE_ENABLED }}/1/g' > dist/ecs-deploy-$$env.yml; \
 	done
 	for env in internal-dev internal-qa; do \
-		cat ecs-proxies-deploy.yml | sed -e 's/{{ SPINE_ENV }}/test/g' -e 's/{{ SANDBOX_MODE_ENABLED }}/0/g' > dist/ecs-deploy-$$env.yml; \
+		cat ecs-proxies-deploy.yml | sed -e 's/{{ SPINE_ENV }}/veit07/g' -e 's/{{ SANDBOX_MODE_ENABLED }}/0/g' > dist/ecs-deploy-$$env.yml; \
 	done
 	cat ecs-proxies-deploy.yml | sed -e 's/{{ SPINE_ENV }}/int/g' -e 's/{{ SANDBOX_MODE_ENABLED }}/0/g' > dist/ecs-deploy-int.yml
-	cat ecs-proxies-deploy.yml | sed -e 's/{{ SPINE_ENV }}/ref/g' -e 's/{{ SANDBOX_MODE_ENABLED }}/0/g' > dist/ecs-deploy-ref.yml
+	cat ecs-proxies-deploy.yml | sed -e 's/{{ SPINE_ENV }}/int/g' -e 's/{{ SANDBOX_MODE_ENABLED }}/0/g' > dist/ecs-deploy-ref.yml
```
Comment (mfjarvis) [inline Makefile:35]: Isn't ref spine separate to int spine?

File: coordinator/src/services/communication/live-spine-client.ts [hunk 2]
```diff
@@ -135,6 +128,7 @@ export class LiveSpineClient implements SpineClient {
 
   private getSpineUrlForPrescription() {
     if (this.spineEndpoint.includes("ref")) {
+      // don't need to do this anymore  ( urls are correct ) also search should be  /^msg\./ (not a global replace)
```
Comment (mfjarvis) [inline coordinator/src/services/communication/live-spine-client.ts:131]: Fix rather than commenting? :wink:

Comment (mfjarvis) [inline coordinator/src/services/communication/live-spine-client.ts:131]: Any reason not to make this change?

File: coordinator/src/services/communication/live-spine-client.ts [hunk 3]
```diff
@@ -11,14 +11,7 @@ const SPINE_ENDPOINT = process.env.SPINE_URL
...
 
-const httpsAgent = new https.Agent({
-  cert: process.env.CLIENT_CERT,
-  key: process.env.CLIENT_KEY,
-  ca: [
-    process.env.ROOT_CA_CERT,
-    process.env.SUB_CA_CERT
-  ]
-})
+const httpsAgent = new https.Agent()
```
Comment (mfjarvis) [inline coordinator/src/services/communication/live-spine-client.ts:14]: Presumably don't even need to pass in an agent anymore, we can just use the default one

File: coordinator/src/services/communication/live-spine-client.ts [hunk 4]
```diff
@@ -36,20 +25,8 @@ export class LiveSpineClient implements SpineClient {
   }
 
   async send(spineRequest: SpineRequest, logger: Logger): Promise<SpineResponse<unknown>> {
-    let wrappedMessage
-    try {
-      logger.info("Building EBXML wrapper for SpineRequest")
-      wrappedMessage = this.ebXMLBuilder(spineRequest, logger)
-    } catch (error) {
-      if (error instanceof FhirMessageProcessingError) {
```
Comment (mfjarvis) [inline coordinator/src/services/communication/live-spine-client.ts:44]: This error formatting is already done in app.ts

File: coordinator/src/services/communication/live-spine-client.ts [hunk 5]
```diff
@@ -134,18 +111,10 @@ export class LiveSpineClient implements SpineClient {
   }
 
   private getSpineUrlForPrescription() {
-    if (this.spineEndpoint.includes("ref")) {
```
Comment (mfjarvis) [inline coordinator/src/services/communication/live-spine-client.ts:137]: The platform-wide egress config now contains the correct ref URLs

File: coordinator/src/services/communication/live-spine-client.ts (hunk 2 above)
Comment (mfjarvis) [reply coordinator/src/services/communication/live-spine-client.ts]: Done

File: coordinator/src/services/communication/live-spine-client.ts (hunk 3 above)
Comment (mfjarvis) [reply coordinator/src/services/communication/live-spine-client.ts]: Done

**Comment by mfjarvis** [comment]
ADO doesn't like this PR :shrug:

File: coordinator/tests/services/translation/request/dispense/release.spec.ts [hunk 6]
```diff
@@ -14,21 +24,9 @@ describe("translateReleaseRequest", () => {
   }])
   const translatedRelease = translateReleaseRequest(parameters, logger)
```
Comment (dan-jones-iw) [inline coordinator/tests/services/translation/request/dispense/release.spec.ts:25]: ugly

Comment (mfjarvis) [reply coordinator/tests/services/translation/request/dispense/release.spec.ts]: Sorted

**Review by dan-jones-iw — APPROVED** [review]

**Pull request merged**